### ISSUE TITLE
[IMPROVEMENTS] Add possibility to add custom/extra logs when crashes are saved

### ISSUE THREAD

**Issue by liszto** [microsoft/appcenter-sdk-unity#335]
Hello,
it could be a really huge improvement for us to add some possibility to add extra infos during the crash saving phase not only during upload process (Attachement system).

With HockeyApp we had implemented this as an extra step but it could be nice to have it natively in the plugin with AppCenter. We made some method abstract and we override them when it was wanted.

Or just extend the attachment system to this moment specifically.

This allow us to bring some extra information like this : 

```
protected override List<string> GetLogHeaders()
{
	List<string> list = base.GetLogHeaders();
	if (ApplicationManager.Instance != null)
	{
		ApplicationManager.Instance.UpdateHockeyAppLogHeaders(list);
	}
	return list;
}
```

With this override, this allow us to obtain a log with "information" like the user in our system, what action he did before this occured etc...

**Comment by Jamminroot** [comment]
Hey, @liszto, thanks for bringing this to our attention. I'll address your request to our PMs team.

**Comment by liszto** [comment]
Yes I read this but as I said it's not what we need in our case :/

**Comment by liszto** [comment]
As a reminder, this is the method that we used with HockeyApp to send custom log to the backend : 

```
/// <summary>
	/// Collect all header fields for the custom exception report.
	/// </summary>
	/// <returns>A list which contains the header fields for a log file.</returns>
	protected virtual List<string> GetLogHeaders ()
	{
		List<string> list = new List<string> ();

		#if (UNITY_ANDROID && !UNITY_EDITOR)
		using (var pluginClass = new AndroidJavaClass(JAVA_HOCKEYUNITYPLUGIN_CLASS))
		{
			var versionCode = pluginClass.CallStatic<string>("getVersionCode");
			var versionName = pluginClass.CallStatic<string>("getVersionName");
			var manufacturer = pluginClass.CallStatic<string>("getManufacturer");
			var model = pluginClass.CallStatic<string>("getModel");
			var deviceIdentifier = pluginClass.CallStatic<string>("getDeviceIdentifier");

			list.Add("Package: " + packageID);
			list.Add("Version Code: " + versionCode);
			list.Add("Version Name: " + versionName);

			var versionComponents = SystemInfo.operatingSystem.Split('/');
			var osVersion = "Android: " + versionComponents[0].Replace("Android OS ", "");
			list.Add (osVersion);

			list.Add("Manufacturer: " + manufacturer);
			list.Add("Model: " + model);
			list.Add("CrashReporter Key: " + deviceIdentifier);
			list.Add("Date: " + DateTime.UtcNow.ToString("ddd MMM dd HH:mm:ss {}zzzz yyyy").Replace("{}", "GMT"));
		}
		#endif

		return list;
	}
```
@annakocheshkova maybe she's knowing more on this feature in appcenter ? :) 
It's almost impossible to switch on AppCenter without this one for some of our project.

**Comment by Jamminroot** [comment]
Just wanted to clarify this :) Apologies for double-checking. I'll let our PMs team look at this.

**Comment by winnieli1** [comment]
Hi @liszto thanks for the request! This isn't on our short term roadmap but I'd like to better understand your use case. 

What types of logs/attachments are you looking to save? And are these attached to crash reports or just standalone logs? Would you be abe to use [custom events](https://docs.microsoft.com/en-us/appcenter/diagnostics/features#events-before-a-crash) to understand the user actions before a crash?

Just want to make sure I fully understand your use case. Thanks!

**Comment by liszto** [comment]
Those logs/attachment are custom string ones. And those ones are attached to crash report.
The display was like this on HockeyApp side : 
![image](https://user-images.githubusercontent.com/408031/66464443-4ae86080-ea7f-11e9-935d-2e7084961f3b.png)

The red part in the log uploaded on the platform is our custom header part.

It's not custom some sort of custom event, it's more custom data that we retrieve during the crash.
Where was the user (which menu, ids, etc...)

**Comment by liszto** [comment]
this is to allow us to understand the crash when the user just say : 
My app crashes during my test yesterday night and he's not able to give us more information on the menu or stuff like this.
And this often happens that's why it's needed for us :/

**Comment by winnieli1** [comment]
Ok got it, thanks for the explanation.  I'm going to move this issue to the App Center repo to track community interest. We don't have immediate plans to address this but you can follow our roadmap to see what our team is working on and I will also post here if I have any updates. 

Thanks!
